reference/src/glossary.md

```diff
@@ -1,5 +1,43 @@
 ## Glossary
 
+#### Aliasing
+
+(Please note: a full aliasing model for Rust has not yet been constructed, but
+at the moment we can give the following definition.)
+
+*Aliasing* is any time one pointer or reference points to a "span" of memory
+that overlaps with the span of another pointer or reference. A span of memory is
+similar to how a slice works: there's a base byte address as well as a length in
+bytes.
```

Comment:
Thanks a lot! I like it.

There is one possible source of confusion here though: LLVM's `noalias` actually uses a very different definition. That one is about the *accesses* that happen through a pointer and whether they *conflict* in the sense that the ordering is relevant. So two pointers that point to immutable memory never alias each other because two read accesses never "conflict".

But we probably want the "standard" notion of aliasing here for Rust, it works better with our usual statement that "shared references may alias". But it might be worth having a note somewhere pointing out that LLVM uses a different definition?